### ISSUE TITLE
MC creation from system after it was used by another MC creates phantom events 

### ISSUE THREAD

**Issue by KiK0S** [osukhoroslov/dslab#328]
mp-kv-replication snippet that produces error

```rust
    let mut mc = ModelChecker::new(&sys);

    let key = random_string(8, &mut rand).to_uppercase();
    let replicas = key_replicas(&key, &sys);
    let non_replicas = key_non_replicas(&key, &sys);
    println!("Key {} replicas: {:?}", key, replicas);
    println!("Key {} non-replicas: {:?}", key, non_replicas);

    // stage 1: get key from the first replica (during the network partition)
    let stage1_strategy = mc_query_strategy(&replicas[0], McQuery::Get(key.clone(), vec![]));
    let stage1_msg = Message::json("GET", &GetReqMessage { key: &key, quorum: 2 });
    let stage1_states = run_mc(
        &mut mc,
        stage1_strategy,
        &replicas[0],
        stage1_msg,
        McNetworkChange::Partition([
            vec![
                replicas[0].clone(),
                non_replicas[0].clone(),
                non_replicas[1].clone(),
                non_replicas[2].clone(),
            ],
            vec![replicas[1].clone(), replicas[2].clone()],
        ]),
        None,
    )?
    .collected_states;
    println!("stage 1: {}", stage1_states.len());
    if stage1_states.is_empty() {
        return Err("stage 1 - GET response is not received".to_owned());
    }

    // stage 2: put key from the first replica (network partition still exists)
    let value = random_string(8, &mut rand);
    mc = ModelChecker::new(&sys);

    let stage2_strategy = mc_query_strategy(&replicas[0], McQuery::Put(key.clone(), vec![value.clone()]));
    let stage2_msg = Message::json(
        "PUT",
        &PutReqMessage {
            key: &key,
            value: &value,
            quorum: 2,
            context: None,
        },
    );
    let stage2_states = run_mc(
        &mut mc,
        stage2_strategy,
        &replicas[0],
        stage2_msg,
        McNetworkChange::Partition([
            vec![
                replicas[0].clone(),
                non_replicas[0].clone(),
                non_replicas[1].clone(),
                non_replicas[2].clone(),
            ],
            vec![replicas[1].clone(), replicas[2].clone()],
        ]),
        None,
    )?
    .collected_states;
    println!("stage 2: {}", stage2_states.len());
    if stage2_states.is_empty() {
        return Err("stage 2 - PUT response is not received".to_owned());
    }

```

stage 2 gets data from stage 1

**Comment by KiK0S** [comment]
looks like it is related to BFS implementation

**Comment by osukhoroslov** [comment]
Fixed in #345
